packages/flutter/lib/src/material/time.dart

```diff
@@ -48,20 +48,21 @@ enum DayPeriod {
 ///  * [DateTime], which represents date and time, and is subject to eras and
 ///    time zones.
 @immutable
-class TimeOfDay {
+class TimeOfDay implements Comparable<TimeOfDay> {
   /// Creates a time of day.
   ///
   /// The [hour] argument must be between 0 and 23, inclusive. The [minute]
   /// argument must be between 0 and 59, inclusive.
-  const TimeOfDay({ @required this.hour, @required this.minute });
+  const TimeOfDay({ @required this.hour, @required this.minute })
+      : assert(hour != null && hour >= 0 && hour < hoursPerDay
+      && minute != null && minute >= 0 && minute < minutesPerHour);
```

Comment:
Nit: might be more readable if these were broken into more asserts (at least one per param?)